src/atomic.cr

```diff
@@ -222,3 +222,34 @@ struct Atomic(T)
     end
   end
 end
+
+# An atomic flag, that can be set or not.
+#
+# Concurrency safe. If many fibers try to set the atomic in parallel, only one
+# will succeed.
+#
+# Example:
+# ```
+# flag = Atomic::Flag.new
+# flag.test_and_set # => true
+# flag.test_and_set # => false
+# flag.clear
+# flag.test_and_set # => true
+# ```
+struct Atomic::Flag
```

Comment:
Shouldn't this be a Class instead, in case such a flag would be passed around?

Reply:
`Atomic(T)` is a struct.